src/rdkafka.c

```diff
@@ -5077,6 +5077,38 @@ rd_kafka_Uuid_t *rd_kafka_Uuid_copy(const rd_kafka_Uuid_t *uuid) {
         return copy_uuid;
 }
 
+/**
+ * Returns a new non cryptographically secure UUIDv4 (random).
+ *
+ * @return A UUIDv4.
+ *
+ * @remark Must be freed after use using rd_kafka_Uuid_destroy().
+ */
+rd_kafka_Uuid_t *rd_kafka_Uuid_random() {
+        int i;
+        unsigned char rand_values_bytes[16] = {};
+        uint64_t *rand_values_uint64        = (uint64_t *)rand_values_bytes;
+        unsigned char *rand_values_app;
+        rd_kafka_Uuid_t *ret = rd_kafka_Uuid_copy(&RD_KAFKA_UUID_ZERO);
+        for (i = 0; i < 16; i += 2) {
+                uint16_t rand_uint16 = (uint16_t)rd_jitter(0, INT16_MAX - 1);
+                /* No need to convert endianess here because it's still only
+                 * a random value. */
+                rand_values_app = (unsigned char *)&rand_uint16;
+                rand_values_bytes[i] |= rand_values_app[0];
+                rand_values_bytes[i + 1] |= rand_values_app[1];
+        }
+
+        rand_values_bytes[6] &= 0x0f; /* clear version */
+        rand_values_bytes[6] |= 0x40; /* version 4 */
+        rand_values_bytes[8] &= 0x3f; /* clear variant */
+        rand_values_bytes[8] |= 0x80; /* IETF variant */
```

Comment:
I think we should move this to rdrand.h

Reply:
It generates a Uuid, all the code for a Uuid is here

Reply:
We should refactor it to a `rduuid.h` + `rduuid.c` later

Reply:
Maybe I can move it to a `rduuid.h` and `rduuid.c`

Reply:
I was talking about the generic part related to random bytes generation here.

Reply:
Regarding rdkafka_uuid.h and rdkafka_uuid.c, we can have them but don't do it for now. Let's do it later when we we won't have much conflicts with other ongoing dev work.

Reply:
> I think we should move this to rdrand.h
here the random value is generated in `rd_jitter` the rest is code related to Uuid